Source/WebCore/Modules/reporting/Report.h

```diff
@@ -31,23 +31,55 @@
 
 namespace WebCore {
 
-class Report : public RefCounted<Report> {
-    WTF_MAKE_FAST_ALLOCATED;
+class WEBCORE_EXPORT Report : public RefCounted<Report> {
+    WTF_MAKE_ISO_ALLOCATED(Report);
 public:
-    static Ref<Report> create(const String& type, const String& url, const RefPtr<ReportBody>&);
+    static Ref<Report> create(const AtomString& type, const String& url, RefPtr<ReportBody>&&);
 
-    virtual ~Report();
+    ~Report();
 
-    const String& type() const { return m_type; }
-    const String& url() const { return m_url; }
-    const RefPtr<ReportBody> body() { return m_body; }
+    const AtomString& type() const;
+    const String& url() const;
+    const RefPtr<ReportBody>& body();
+
+    template<typename Encoder> void WEBCORE_EXPORT encode(Encoder&) const;
+    template<typename Decoder> static WEBCORE_EXPORT std::optional<Ref<WebCore::Report>> decode(Decoder&);
 
 private:
-    explicit Report(const String& type, const String& url, const RefPtr<ReportBody>&);
+    explicit Report(const AtomString& type, const String& url, RefPtr<ReportBody>&&);
 
-    String m_type;
+    AtomString m_type;
     String m_url;
     RefPtr<ReportBody> m_body;
```

Comment:
Can ReportBody be empty??? If not, why don't we use Ref instead?

Reply:
According to the spec, The `ReportBody` in a `Report` is nullable, so I think it's appropriate to be a `RefPtr`, though I don't believe we ever construct a `Report` without a `ReportBody`.